bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java

```diff
@@ -2445,19 +2453,21 @@ public Cursor getSystemCursor (int id) {
  */
 @Override
 public Font getSystemFont () {
+	return getSystemFont(getPrimaryMonitor().getZoom());
```

Comment:
As the passed parameter won't be used from the default font factory anyway, this would be only an issue when the "swt.autoScale.updateOnRuntime" flag is set to true, which should not be possible for Windows 7 or 8 anyway due to most of the dpi dependent OS calls that were introduced with Windows 10.